core/src/block_strider/provider/blockchain_provider.rs

```diff
@@ -93,12 +93,12 @@ impl BlockchainBlockProvider {
                             return parsed;
                         }
                     }
-                    None => tracing::warn!(?prev_block_id, "block not found"),
+                    None => {
+                        tracing::warn!(?prev_block_id, "block not found");
```

Comment:
Just return `None`. We should probably log inside the retry provider when it has exceeded its attempts. Otherwise, it creates useless noise